### ISSUE TITLE
Coments are not allowed because of ArtifactFilterParseException

### ISSUE THREAD

**Issue by TobseF** [AlexisJehan/MvnCheck#3]
The _README_ examples look like **commments** are allowed in the `.mvnchk-ignore` file.

But adding a comment like this
``` ini
# Comment
com.google.guava:guava:*-android
```
will throw an `ArtifactFilterParseException`:
```
ArtifactFilterParseException: Unexpected format: "ï»¿" (at line 1 of the .\.mvnchk-ignore file)
```
I don't know if this is a bug or a missing feature. 
Anyway, an explanation comment in the `.mvnchk-ignore` would be very helpful. 
For example, to explain why we have to stick to a specific version.

**Comment by TobseF** [comment]
I double checked the steps: Just create a new text file as .mvnchk-ignore in IntelliJ. It's created in UTF-8. Cop the first two lines text from the README. -> It will fail.

It's the `#` character that causes the problem:

#### This will work fine:
```
com.google.guava:guava
com.google.guava:guava
```
#### Adding the `#` will break it  `⚡ Unexpected format: "ï»¿"`
```
# com.google.guava:guava
com.google.guava:guava
```

![image](https://github.com/user-attachments/assets/23ccced7-0123-4b1d-a41d-246ffbe78a05)

This is the file (renamed - otherwise GitHub won't accept):
 [📑 mvnchk-ignore.txt](https://github.com/user-attachments/files/16533575/mvnchk-ignore.txt)

But I'm sure it has nothing to do with UTF-8 encoding or line breaks. By the way... running it on Windows in the cmd bash.

Hope this helps a little bit to track the 🐞

**Comment by AlexisJehan** [comment]
I'm currently on vacation, I'll investigate the issue and your merge request once I'll back at home.

**Comment by AlexisJehan** [comment]
Hello back,
I can confirm this is an issue related to the BOM character of the UTF-8 with BOM encoding. I only tried with UTF-8 (without BOM) files so I never had the issue until now. The BOM character was not ignored and interpreted before the `#` character of the comment line, that was the source of the exception.

**Comment by AlexisJehan** [comment]
Released in [1.7.0](https://github.com/AlexisJehan/MvnCheck/releases/tag/v1.7.0).
